pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -330,6 +331,9 @@ stdenv.mkDerivation ({
         export NIX_${ghcCommandCaps}="${ghcEnv}/bin/${ghcCommand}"
         export NIX_${ghcCommandCaps}PKG="${ghcEnv}/bin/${ghcCommand}-pkg"
         export NIX_${ghcCommandCaps}_DOCDIR="${ghcEnv}/share/doc/ghc/html"
+        export LD_LIBRARY_PATH="''${LD_LIBRARY_PATH:+''${LD_LIBRARY_PATH}:}${
+          concatStringsSep ":" (map (x: x + "/lib") (filter isDerivation systemBuildInputs))
```

Comment:
Yes, filtering for `!= null` is probably the way to go for the time being. The code that constructs those lists needs cleaning up anyway. At some point I'll tackle that task and then I'll make sure those `null` values don't show up anymore.